tests/test_zarr_ipfs.py

```
@@ -137,3 +137,43 @@ def test_encryption(random_zarr_dataset: tuple[str, xr.Dataset]):
+    # Now trying to load without a decryptor, xarray should be able to read the metadata and still perform operations on the unencrypted variable
+    print("Attempting to read and print metadata of partially encrypted zarr")
+    ds = xr.open_zarr(
+        store=HAMT(store=IPFSStore(          
+            api_key="test",
+            basic_auth=("test", "test")
+        ), root_node_id=hamt.root_node_id, read_only=True)
+    )
+    print(ds)
+    assert ds.temp.sum() == expected_ds.temp.sum()
+    # We should be unable to read precipitation values which are still encrypted
+    with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)